snappiershot/serializers/json.py

```diff
@@ -100,8 +103,11 @@ def encode_numeric(value: Number) -> JsonType:
             # These types are by default supported by the JSONEncoder base class.
             return value
         if isinstance(value, complex):
-            encoded_value = [value.real, value.imag]
+            encoded_value: List[float] = [value.real, value.imag]
             return CustomEncodedNumericTypes.complex.json_encoding(encoded_value)
+        if isinstance(value, Decimal):
+            encode_value: List[Any] = list(value.as_tuple())
```

Comment:
One thing to consider:
I just ran this locally and the "encoded" `Decimal("3.1415")` is `[0, [3, 1, 4, 1, 5], -4]` which isn't very informative... Although this works, I think a nicer format for this data would be converting this named tuple to a dictionary, which is simple to do.

So in the `encode_numeric` method you would have `encode_value = value.as_tuple()._asdict()`, and in the `decode_numeric` method you would have `return Decimal(DecimalTuple(**value))`. This would make the encoding of `Decimal("3.1415")` be `{"sign": 0, "digits": [3, 1, 4, 1, 5], "exponent": -4}` which is more verbose but much easier to understand.

Reply:
Great idea.